src/System.Globalization/tests/NumberFormatInfo/NumberFormatInfoCurrentInfo.cs

```diff
@@ -20,46 +21,35 @@ public static IEnumerable<object[]> CurrentInfo_CustomCulture_TestData()
         [MemberData(nameof(CurrentInfo_CustomCulture_TestData))]
         public void CurrentInfo_CustomCulture(CultureInfo newCurrentCulture)
         {
-            CultureInfo oldCurrentCulture = CultureInfo.CurrentCulture;
-            try
+            RemoteInvoke((cultureCode) =>
             {
-                CultureInfo.CurrentCulture = newCurrentCulture;
-                Assert.Same(newCurrentCulture.NumberFormat, NumberFormatInfo.CurrentInfo);
-            }
-            finally
-            {
-                CultureInfo.CurrentCulture = oldCurrentCulture;
-            }
+                CultureInfo newCulture = new CultureInfo(cultureCode);
+                CultureInfo.CurrentCulture = newCulture;
+                Assert.Same(newCulture.NumberFormat, NumberFormatInfo.CurrentInfo);
+                return SuccessExitCode;
+            }, newCurrentCulture.TwoLetterISOLanguageName).Dispose();
```

Comment:
This test is failing CI for Windows but I can't repro it on my local machine. I'm wondering if it's a concurrency issue, though it failed for both dbg and ret and I don't see where the issue would be.